proposals/0002-root-signature-in-clang.md

```diff
@@ -402,23 +402,75 @@ registers are bound multiple times, or where there are multiple RootFlags
 entries, so subsequent stages should not assume that any given root signature in
 IR is valid.
 
+#### Metadata Schema
 
-> **Open Question**: what should the named metadata be?  There's options I
-> think...
+* RootConstant
+
+| ParameterType | ShaderVisibility | ShaderRegister | RegisterSpace | Num32BitValues |
+|---------------|------------------|----------------|---------------|----------------|
+| String        | int              | int            | int           | int            |
+
+
+* RootDescriptor
+
+| ParameterType | ShaderVisibility | ShaderRegister | RegisterSpace | RootDescriptorFlags |
+|---------------|------------------|----------------|---------------|---------------------|
+| String        | int              | int            | int           | int                 |
+
+
+* DescriptorRange
+
+| RangeType | NumDescriptors | BaseShaderRegister | RegisterSpace | DescriptorRangeFlags |
+|-----------|----------------|--------------------|---------------|----------------------|
+| String    | int            | int                | int           | int                  |
+
+
+* DescriptorTable
+
+| DescriptorRanges          |
+|---------------------------|
+| list of DescriptorRanges  |
+
+* StaticSampler
+  
+| Filter | AddressU | AddressV | AddressW | MipLODBias | MaxAnisotropy | ComparisonFunc | BorderColor | MinLOD | MaxLOD | ShaderRegister | RegisterSpace | ShaderVisibility |
+|--------|----------|----------|----------|------------|---------------|----------------|-------------|--------|--------|----------------|---------------|------------------|
+| int    | int      | int      | int      | float      | int           | int            | int         | float  | float  | int            | int           | int              |
+
+* RootSignature
+
+| RootFlag | Parameters                                      | StaticSamplers          |
+|----------|-------------------------------------------------|-------------------------|
+| int      | list of RootDescriptor/RootConstant/DescriptorTable | list of StaticSamplers |
```

Comment:
This doesn't match the example.  I think really it should be a list of parameters, and each parameter could be a RootDescriptor, RootConstant, DescriptorTable or StaticSampler.  I don't think there's any requirement for the static samplers to come after the other parameters - the grammar doesn't enforce that, so I'm not sure why we would need to add that extra requirement to the metadata format?

Reply:
Updated the example.
The list will be null if no static samplers.

Reply:
I don't understand why these two lists are separated out.  They're not separate in the grammar, so why are we introducing that here?

If this is to match the d3d12 struct definitions, I don't think that's a goal here.

Reply:
It started from 2 different list in the string version of root signature.
And will end to 2 different list in the final serialized root signature.
It will be strange to merge them into one list and split later.

Reply:
Note also that in the example StaticSampler starts with a parameter type, "StaticSampler", precisely so that it can be intermingled with the other parameters.

Reply:
My mistake, I though Static Sampler have to be at end of the root signature which is not true.
If we want to keep things in the same order of the string version of root signature.
Then make a one list make sense.